src/eduid_idp/assurance.py

```diff
@@ -103,22 +109,29 @@ def __init__(self, user, sso_session, logger):
                         self.swamid_al2_used = True
                     elif cred.proofing_method == METHOD_SWAMID_AL2_MFA_HI:
                         self.swamid_al2_hi_used = True
-        self.is_swamid_al2 = False
+            # External mfa check
+            if sso_session.external_mfa is not None:
+                self.logger.debug('External MFA issuer: {}'.format(sso_session.external_mfa.issuer))
+                self.external_mfa_used = True
+                # TODO: Support more SwedenConnect authn contexts?
+                if sso_session.external_mfa.authn_context == 'http://id.elegnamnden.se/loa/1.0/loa3':
+                    self.swamid_al2_hi_used = True
+
         if user.nins.verified.to_list():
             self.is_swamid_al2 = True
 
     def __str__(self):
         return (f'<AuthnState: creds={len(self._creds)}, pw={self.password_used}, fido={self.fido_used}, '
-                f'nin is al2={self.is_swamid_al2}, token is (al2={self.swamid_al2_used}, '
-                f'al2_hi={self.swamid_al2_hi_used})>')
+                f'external_mfa={self.external_mfa_used}, nin is al2={self.is_swamid_al2}, '
+                f'mfa is (al2={self.swamid_al2_used}, al2_hi={self.swamid_al2_hi_used})>')
 
     @property
     def is_singlefactor(self):
         return self.password_used or self.fido_used
 
     @property
     def is_multifactor(self):
-        return self.password_used and self.fido_used
+        return (self.password_used and self.fido_used) or (self.password_used and self.external_mfa_used)
```

Comment:
would prefer self.password_used and (self.fido_used or self.external_mfa_used) - easier to follow